docs/ota-client-guide/modules/ROOT/pages/garage-sign.adoc

```diff
@@ -0,0 +1,218 @@
++
+For example, when metadata is edited, the new metadata needs to be signed by `garage-sign` to become valid. The new `targets` key generated is used to sign the new `targets.json` file and has the following format:
++
+[source,bash]
+----
+garage-sign targets sign \
+  --repo <reponame> \
+  --key-name <keyname>
+----
++
+An example of this can be seen in xref:customise-targets-metadata.adoc#_anatomy_of_targets_json_metadata[adding custom metadata fields to targets metadata].
+
+
+* *Sign metadata expiry date*
++
+You can also define an expiry date for your metadata using `garage-sign` on the command line. You can do this in two ways, but only one type of expiry argument must be used to define the metadata expiry date.
+
+** *Defining a fixed date and time for metadata expiry*
++
+Use the following format of `garage-sign` to define a specific time in UTC for metadata to expire:
++
+[source,bash]
+----
+garage-sign targets sign \
+  --expires 2018-01-01T00:01:00Z \ <1>
+  --repo <reponame> \
+  --key-name <new_targets_keyname>
+----
++
+<1> This is an example of specifying a date and time of expiry as an intance in UTC. After this fixed date and time, metadata will no longer be valid.
+
+** *Defining a period of time for metadata expiry*
++
+Use the following format of `garage-sign` to specify the period of time for metadata to be valid:
++
+[source,bash]
+----
+garage-sign targets sign \
+  --expire-after 1Y3M5D \ <1>
+  --repo <reponame> \
+  --key-name <new_targets_keyname>
+----
++
+See our guide xref:metadata-expiry.adoc[managing metadata expiry dates] for more information.
++
+<1> This is an example of specifying a period of time for expiry of metadata. The expiration period is defined in the order of years, months and days (each optional but in this specified order).
+
+
+== Add and Remove software from repositories
+
+* *Add software*
++
+OTA Connect also gives you the opportunity to add software by using the command `garage-sign targets add`.
++
+[source,bash]
+----
+garage-sign targets add \
+  --repo <reponame> \
+  --format <ostree | binary> \ <1>
+  --length <imagelength> \ <2>
+  --name <keyname> \
+  --version <version> \ <3>
+  --sha256 <imagehash> \ <4>
+  --hardwareids <hwid> \
+  --url <softwareurl> <5>
+----
++
+<1> You must specify the format of the target image to be added--either as an ostree or binary image. By convention, in OTA Connect, the `garage-sign` tool is used to upload binary images because OSTree images can be uploaded directly in the OTA Connect Portal. Thus in most cases the format of images to be uploaded to your OTA Connect account using this tool is `binary`.
+<2> The length of the image to be added must be specified in bytes.
+<3> You must also specify the version number of the target image.
+<4> The sha256 hash of the software image is specified here.
+<5> By convention, in OTA Connect, the target image must first be uploaded via a URL before it can be added. The URL must be specified here.

```

Comment:
This isn't "by convention", and "uploaded via a URL" doesn't mean anything.

When aktualizr tries to download a file, it first looks for it at a specific place in the repo based on its name. But when you use `garage-sign targets add`, you're just putting new metadata in place--you still need to tell aktualizr where it can find the file that the metadata is talking about. Since `garage-sign` doesn't upload the file for you, you need to upload it somewhere yourself, and provide garage-sign with a URL where it's available so that aktualizr can go get it.